addons/stock/report/report_deliveryslip.xml

```diff
@@ -225,8 +229,20 @@
                 <span t-esc="description"/>
             </p>
         </td>
-        <t t-if="has_serial_number" name="move_line_lot">
-            <td><span t-field="move_line.lot_id.name"/></td>
+        <t t-if="has_serial_number">
+            <td name="move_line_lot">
```

Comment:
I agreed your version is cleaner/more logical but I would let the `name` attribute where it was before to not break existing xpath (and thus, no need to update `product_expiry` code)
```suggestion
        <t t-if="has_serial_number" name="move_line_lot">
            <td>
```
That said, you can make this change on `master`, when the PR will be forward-ported